### ISSUE TITLE
Support for C++/CLI (managed C++)

### ISSUE THREAD

**Issue by gwtov** [dotnet/roslyn-analyzers#6252]
Hi! I'm not sure if this is the correct place for this question, sorry if this is not the case!

It seems like the analyzers do not support C++/CLI code.

For example, when running a code analysis on a **CLR project that targets .NET Framework 4.7.2** a violation of 
rule [CA1001](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1001) is detected for this example:

```
public ref class NoDisposeMethod
{
    public:
        System::IO::FileStream^ _newFile;

        NoDisposeMethod()
        {
            _newFile = gcnew System::IO::FileStream("c:\\temp.txt", System::IO::FileMode::Open);
        }
};
```

However, no violation is detected for the same example in a **CLR project that targets .NET 6**.

Does this have something to do with the switch from FxCop analyzers to Roslyn analyzers?

**Comment by Evangelink** [comment]
Hi @gwtov,

You are right. The old FxCop tool was working by analyzing the binary (by doing so it was agnostic to language - as long as it's CLR) while roslyn analyzers rely on roslyn which is the C#/VB.NET compiler. So there is no support for C++/CLI.
